### ISSUE TITLE
QBittorent run external on completion not working

### ISSUE THREAD

**Issue by ayostepht** [cross-seed/cross-seed#568]
I've been beating my head trying to troubleshoot this issue and I can't for the life of me find a solution.

I'm trying to get cross-seed to automatically run on torrent completion in Binhex's QbittorrentVPN on Unraid.  I have the following in "run external programs" in qBit: curl -XPOST http://192.168.0.115:2468/api/webhook?apikey=[redacted] --data-urlencode "infoHash=[redacted]" and I have port 2468 set as VPN output port in the QBitVPN container.  I'm certain my APIkey is correct (I have it running correctly in Autobrr). 

 I'm seeing the following in the qbit logs:

2023-12-13T12:38:34 - Torrent download finished. Torrent: "Varsity.Blues.1999.1080p.BluRay.x264-nikt0"

2023-12-13T12:38:34 - Running external program. Torrent: "Varsity.Blues.1999.1080p.BluRay.x264-nikt0". Command: `curl -XPOST http://192.168.0.115:2468/api/webhook?apikey=[redacted] --data-urlencode "infoHash=[redacted]"` 

But I'm not seeing anything regarding this action in the cross-seed logs. I'm quadruple checked my API key and it is correct. The logs show autobrr's cross-seed API calls working correctly as well.

**Comment by zakkarry** [comment]
https://www.cross-seed.org/docs/basics/faq-troubleshooting#i-cant-reach-my-prowlarrjackettcross-seedtorrent-client-using-dockervpn

It's mentioned here...you're trying to connect to your LAN IP address from a VPN container. You will either need to configure it properly or setup a custom network and set that up properly for this.

**Comment by Sapkotakx** [comment]
I am also having same issue. I have passed lan network on Qbit config. I tried using ip of cross-seed and “cross-seed” but no luck.

**Comment by ayostepht** [comment]
I was able to get it the announcement from QBit to Cross-seed to work with the help of someone on Reddit yesterday by passing the public IP address of the QBit container into the curl command. 

Now, I'm having a separate issue. I've tried all 3 ways of matching (name, data, hash) and each one doesn't work quite right for me. 
**Hash error:** 
"2023-12-13 18:06:10 [32minfo[39m: [server] Received search request: { infoHash: 'c81504a7c28a9345903b95ada96926b32183385a' }
2023-12-13 18:06:10 [36mverbose[39m: [prefilter] Torrent Jurassic.World.2015.BDRip.x264-SPARKS was not selected for searching because not all files are videos
2023-12-13 18:06:10 [32minfo[39m: [server] Did not search for { infoHash: 'c81504a7c28a9345903b95ada96926b32183385a' "

**Data error:**
023-12-13 18:19:24 [32minfo[39m: [server] Received search request: {
  path: '/media/torrents/Free Leach/Pinocchio.2022.2160p.UHD.BluRay.x265-B0MBARDiERS'
}
2023-12-13 18:19:25 [36mverbose[39m: [prefilter] Torrent Pinocchio.2022.2160p.UHD.BluRay.x265-B0MBARDiERS was not selected for searching because not all files are videos
2023-12-13 18:19:25 [32minfo[39m: [server] Did not search for {
  path: '/media/torrents/Free Leach/Pinocchio.2022.2160p.UHD.BluRay.x265-B0MBARDiERS'
}

**Name error:**
2023-12-13 18:21:21 [32minfo[39m: [server] Received search request: { name: 'Magnum.P.I.2018.S05E18.720p.HDTV.x264-SYNCOPY"' }
2023-12-13 18:21:21 [31merror[39m: could not find a torrent with the criteria { name: 'Magnum.P.I.2018.S05E18.720p.HDTV.x264-SYNCOPY"' }

I have passed my data dir in my config and data based searching is working in general, and my BT_backups folder is passed through. Any thoughts?

**Comment by zakkarry** [comment]
Yes, my immediate thoughtbis that you are not reading the docs provided.

includeNonVideo neess to be set to true.

This is in the config, as well as docs.

**Comment by Sapkotakx** [comment]
@ayostepht, if you passed public IP and got it to work, how will it work if the ip changes. Could you please share your knowledge how did you do it?  I may be missing something. Doco say about splitting tunnelling and stuffs but haven’t tried it since I don’t have my cross-seed exposed.

@zakkarry bro, i have read multiple times and couldn’t figure it out therefore I am seeing assistance here. If you have are going to say to “read” in every post then you are not helping. Please move on else share your experience. Don’t be dh

**Comment by ayostepht** [comment]
@Sapkotakx I think I was mistaken, the ip address I'm using is a 172 which I assumed was public because all of my internal addresses are 192, but some quick research shows that that is probably a private IP (but not from the internal network that I use to access my containers, etc...). When I run IP A within the cross-seed container, I'm using the ETH IP address provided. It is set to lifetime preferred, so I'm hopeful it will stay static. So far the logs are showing it is still working correctly. 

Also, FWIW, since my last comment I figured out my issue, and it wasn't from lack of reading the docs (I have read them over and over...), I accidentally had an extra " on the end of the run external programs command.

**Comment by Sapkotakx** [comment]
I saw your post in reddit, someone recommended to use 172.xx container’s private ip. Do you need to pass cross-seed port on qbit? 
I am using custom network hopefully I won’t face any issues.

**Comment by ayostepht** [comment]
I have the CS port set as VPN output port in Qbit, but I'm not sure its necessary. I added that in my troubleshooting before the 172 address, but didn't remove it. Since it is working now I don't want to poke the bear.

**Comment by Sapkotakx** [comment]
It used to work with 192 address before I updated cross-seed. I will try without passing and see if it works. Thanks for your help. I appreciate it.

**Comment by zakkarry** [comment]
> @zakkarry bro, i have read multiple times and couldn’t figure it out therefore I am seeing assistance here. If you are going to say to “read” in every post then you are not helping. Please move on else share your experience. Don’t be dh

I spent many hours writing/explaining in the docs. I did it for a reason.

https://www.cross-seed.org/docs/basics/options#includenonvideos

The error messages tells you why it's not cross-seeding that specific item.

The config file also describes specifics about each thing. 

If you expect a developer of a project to not point you to the documentation where they've described it in detail for that exact purpose, I don't know what to tell you.

Those docs are me sharing my experience. I'm not the one being a "dh" here.
